### PR TITLE
Sanitise SearchableText when querying catalog in search and filter

### LINKED ISSUE #85: Searching for 'and' followed by a space and text causes exception
# The issue
Searching for the word 'and' followed by a space and more text (e.g. "and myWord") causes the following exception to occur:

ParseError: Token <Token:ATOM> required, u'and' found

## Steps to reproduce
- Add a collection search portlet
- In the search, search for "and test"

## Expected result
- The results for the string "and test" are returned

## Actual result
- A ParseError exception is thrown

# Stack Trace
```
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 155, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 337, in publish_module
  Module ZPublisher.WSGIPublisher, line 255, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 61, in call_object
  Module zope.browserpage.simpleviewclass, line 41, in __call__
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 61, in __call__
  Module zope.pagetemplate.pagetemplate, line 135, in pt_render
  Module Products.PageTemplates.engine, line 88, in __call__
  Module z3c.pt.pagetemplate, line 173, in render
  Module chameleon.zpt.template, line 306, in render
  Module chameleon.template, line 209, in render
  Module chameleon.template, line 187, in render
  Module 45d862eed365f94eae40d06a0f3d077a.py, line 1769, in render
  Module 377a257effc24713ae2f4b0305d3d73e.py, line 1028, in render_master
  Module 377a257effc24713ae2f4b0305d3d73e.py, line 345, in render_content
  Module 45d862eed365f94eae40d06a0f3d077a.py, line 1759, in __fill_content_core
  Module 45d862eed365f94eae40d06a0f3d077a.py, line 509, in render_content_core
  Module 45d862eed365f94eae40d06a0f3d077a.py, line 266, in render_listing
  Module Products.PageTemplates.expression, line 101, in __call__
  Module plone.app.contenttypes.browser.collection, line 47, in batch
  Module plone.app.contenttypes.browser.collection, line 42, in results
  Module plone.app.contenttypes.behaviors.collection, line 121, in results
  Module plone.app.querystring.querybuilder, line 98, in __call__
  Module plone.app.querystring.querybuilder, line 170, in _makequery
  Module Products.CMFPlone.CatalogTool, line 457, in searchResults
  Module Products.ZCatalog.ZCatalog, line 611, in searchResults
  Module Products.ZCatalog.Catalog, line 1091, in searchResults
  Module Products.ZCatalog.Catalog, line 634, in search
  Module Products.ZCatalog.Catalog, line 564, in _search_index
  Module Products.ZCTextIndex.ZCTextIndex, line 209, in query_index
  Module Products.ZCTextIndex.QueryParser, line 153, in parseQuery
  Module Products.ZCTextIndex.QueryParser, line 193, in _parseOrExpr
  Module Products.ZCTextIndex.QueryParser, line 206, in _parseAndExpr
  Module Products.ZCTextIndex.QueryParser, line 241, in _parseTerm
  Module Products.ZCTextIndex.QueryParser, line 260, in _parseAtom
  Module Products.ZCTextIndex.QueryParser, line 188, in _get
  Module Products.ZCTextIndex.QueryParser, line 174, in _require
ParseError: Token <Token:ATOM> required, u'and' found

 - Expression: "view/batch"
 - Filename:   ... .egg/plone/app/contenttypes/browser/templates/listing.pt
 - Location:   (line 22: col 31)
 - Source:     <tal:results define="batch view/batch;
                                          ^^^^^^^^^^
 - Expression: "provider:plone.abovecontentbody"
 - Filename:   ... egg/Products/CMFPlone/browser/templates/main_template.pt
 - Location:   (line 94: col 70)
 - Source:     ... 
                                     ^
 - Expression: "context/@@main_template/macros/master"
 - Filename:   ... .egg/plone/app/contenttypes/browser/templates/listing.pt
 - Location:   (line 6: col 21)
 - Source:     ... tal:use-macro="context/@@main_template/macros/master"
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x1095ebe10>
               views: <ViewMapper - at 0x1104f1d10>
               modules: <_SecureModuleImporter - at 0x104bc8650>
               args: <tuple - at 0x102f58050>
               here: <ImplicitAcquisitionWrapper testcollection at 0x10c52e050>
               wrapped_repeat: <SafeMapping - at 0x1102c2c20>
               user: <ImplicitAcquisitionWrapper - at 0x10c52ef50>
               nothing: <NoneType - at 0x102e7a718>
               container: <ImplicitAcquisitionWrapper testcollection at 0x10c52e050>
               root: <ImplicitAcquisitionWrapper  at 0x10c59eaf0>
               request: <WSGIRequest - at 0x1111579d0>
               traverse_subpath: <list - at 0x1104663f8>
               default: <object - at 0x102fbb4a0>
               context: <ImplicitAcquisitionWrapper testcollection at 0x10c52e050>
               view: <SimpleViewClass from /Users/jefferson/.buildout/eggs/plone.app.contenttypes-2.1.2-py2.7.egg/plone/app/contenttypes/browser/templates/listing.pt listing_view at 0x110269b90>
               translate: <function translate at 0x1107bdb18>
               macroname: master
               options: {...} (0)
               loop: {...} (0)
               target_language: <NoneType - at 0x102e7a718>
```

### PR DESCRIPTION
**Pull request by JeffersonBledsoe**
Closes #85 . Keywords used in the search were not being quoted, causing exceptions in `CTextIndex.QueryParser`. This PR uses the same fix used in Products.CMFPlone.browser.search to quote the necessary keywords.

Additional changes:
- A couple of flake8 fixes
- Updated travis.yml to fix failling tests

Please squash and merge when this gets merged to master, it took a couple of goes to find the right place to apply this fix and get the tests passing

### CODE REVIEW

**Comment by djay** [comment]
I'm suprised there is not code we can call tondo this already without having to repeat the code here?

**Comment by djay** [comment]
I'm suprised there is not code we can call tondo this already without having to repeat the code here?

File: src/collective/collectionfilter/query.py [hunk 1]
```diff
@@ -7,7 +7,37 @@
...
+        term = '"%s"' % term
+    return term
+
+
+def sanitise_search_query(query):
+    for char in BAD_CHARS:
+        query = query.replace(char, u" ")
+    clean_query = [quote_keywords(token) for token in query.split()]
+    clean_query = quote_unsafe_chars(clean_query)
+    return u" ".join(clean_query)
```
Comment (petschki) [inline src/collective/collectionfilter/query.py]: There is similar code here https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/browser/search.py#L23 ... maybe you can reuse it in some way?

**Comment by JeffersonBledsoe** [comment]
@petschki Would you mind taking a look through this again? If it's good, would you mind squashing & merging rather than a standard merge, this isn't my cleanest of PRs! 😳

**Comment by instification** [comment]
@thet @petschki is it possible to get this PR reviewed/merged?

**Review by petschki — CHANGES_REQUESTED** [review]
Can you rebase this on master in order to get the tests running again?

**Comment by JeffersonBledsoe** [comment]
@petschki Looks like the setup step for the tests wasn't working correctly. Should be fixed now, waiting on the tests!

**Review by petschki — APPROVED** [review]
Lgtm

**Pull request merged**